### ISSUE TITLE
LeakCanary reports a memory leak in Session object

### ISSUE THREAD

**Issue by caltseng** [uservoice/uservoice-android-sdk#198]
From the logs: 

08-06 15:15:03.418   8729-11967/LeakCanary﹕ * GC ROOT static com.uservoice.uservoicesdk.Session.instance
08-06 15:15:03.418   8729-11967/LeakCanary﹕ * references com.uservoice.uservoicesdk.Session.context

It's interesting because it doesn't look like the context is being kept in a static object, which would be a source of memory leaks. Please investigate.

**Comment by austintaylor** [comment]
The Session is a singleton and is intended to be kept in memory. It holds onto a Context in order to have access to SharedPreferences. Is this a problem?

**Comment by caltseng** [comment]
Yes. Context should not be kept as static member variables. Please see the below stack overflow links.

http://stackoverflow.com/questions/3346080/android-references-to-a-context-and-memory-leaks

http://stackoverflow.com/questions/28457499/using-context-as-a-member-variable-in-all-classes



From: Austin Taylor
Reply-To: uservoice/uservoice-android-sdk
Date: Thursday, August 20, 2015 at 11:58 AM
To: uservoice/uservoice-android-sdk
Cc: Caleb Tseng
Subject: Re: [uservoice-android-sdk] LeakCanary reports a memory leak in Session object (#198)


The Session is a singleton and is intended to be kept in memory. It holds onto a Context in order to have access to SharedPreferences. Is this a problem?

**Comment by austintaylor** [comment]
I see. I guess we can just pass the context everywhere. 5629cdc

**Comment by caltseng** [comment]
Yeah…We had to change our code to do this as well to solve our memory leaks. It’s the unfortunate way that Android works, I guess.
